src/high-level/constructors.lisp

```diff
@@ -64,7 +64,8 @@ SHAPE can either be a list of dimensions or a fixnum defining the length of the
 If VALUE is not specified then 1 is used.
 If TYPE is not specified then it is inferred from the type of VALUE, defaulting to *DEFAULT-TENSOR-TYPE*.
 LAYOUT specifies the internal storage representation ordering of the returned tensor.
-The tensor is specialized on SHAPE and TYPE."
+The tensor is specialized on SHAPE and TYPE.
```

Comment:
I know this is daring, but maybe we should implement EYE in terms of FROM-DIAG until somebody complains

Reply:
That would definitely be good to reduce code duplication, however with the current way that `FROM-DIAG` takes in the list, it would be causing more cons-ing.